internal/usecase/repository/hostconf_jwk_repository.go

```diff
@@ -17,77 +17,140 @@ var notImplementedError = fmt.Errorf("TODO: not implemented")
 
 type hostconfJwkRepository struct {
 	config *config.Config
-	// TODO: store JWKs in database
-	signingKeys []jwk.Key
-	publicKeys  []string
 }
 
 func NewHostconfJwkRepository(cfg *config.Config) repository.HostconfJwkRepository {
 	if cfg == nil {
 		panic("'cfg' is nil")
 	}
 	r := &hostconfJwkRepository{config: cfg}
-	err := r.generateEphemeralJWK()
-	if err != nil {
-		panic(err)
-	}
 	return r
 }
 
 // CreateJWK generates and inserts a new JWK into the database. The private
 // key is encrypted with the current app secret.
-func (r *hostconfJwkRepository) CreateJWK(db *gorm.DB) (model *model.HostconfJwk, err error) {
-	return nil, notImplementedError
+func (r *hostconfJwkRepository) InsertJWK(db *gorm.DB, hcjwk *model.HostconfJwk) (err error) {
+	if db == nil {
+		return internal_errors.NilArgError("db")
+	}
+
+	if err = db.Create(&hcjwk).Error; err != nil {
+		return err
+	}
+	return nil
 }
 
 // RevokeJWK revokes a JWK with key identifier `kid`
-func (r *hostconfJwkRepository) RevokeJWK(db *gorm.DB, kid string) (model *model.HostconfJwk, err error) {
-	return nil, notImplementedError
+func (r *hostconfJwkRepository) RevokeJWK(db *gorm.DB, kid string) (hcjwk *model.HostconfJwk, err error) {
+	if db == nil {
+		return nil, internal_errors.NilArgError("db")
+	}
+	// find JKW by unique kid
+	if err = db.
+		Where("key_id = ?", kid).
+		First(&hcjwk).
+		Error; err != nil {
+		return nil, err
+	}
+
+	// Set encrypted JWK field to NULL and safe the new object
```

Comment:
nit: typo safe -> save.

Also, it probably justifies a comment to explain that revocation is accomplished simply
by removing the (encrypted) private key. This was (for me) an unexpected solution.  It's
perfectly fine but probably deserves a brief commentary.

Reply:
I have moved the revocation 'logic' to the hostconf JWK model and added comments to the function and model.

I think it is an elegant solution. We don't need the private key of a revoked JWK anyway. The approach does not need an additional column in the DB and we cannot accidentally use the private key of a revoked JWK.

Reply:
Thanks for that.  I agree it's elegant, and the implementation is easier to understand now.